### PR TITLE
feat: support degradation config for Kitex client

### LINKED ISSUE #6: Proposal: support degradation config for Kitex server
Please help add "degradation" as a new type of service governance config to these extensions, instructing Kitex **clients** to drop the corresponding percentile of requests, which is helpful when the server is confronted with an overwhelming amount of traffic.

For example, the config value may be:

```json
{
    "enabled": true,
    "percentage": 50
}
```

Meaning the server should drop 50% of the traffic.
The default value should be `{"enabled": false}` to avoid loss.

Note that Kitex provides a builtin ACL mechanism by `client.WithACLRules`, which is helpful to implement the degradation policy.

Please prepare a tech plan (lark doc) for this task, including the background, goal, your design details and any problems for discussion before proceeding to the coding phase.

### PR DESCRIPTION
**Pull request by CryoPhotonix**
#### What type of PR is this?

feat: support degradation config for Kitex client
#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [x] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io)

#### (Optional) Translate the PR title into Chinese.
为kitex client支持服务降级特性

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en:
zh(optional): 

#### (Optional) Which issue(s) this PR fixes:

Fixes #6

### CODE REVIEW

File: pkg/utils/options.go
Comment (felix021) [inline pkg/utils/options.go:1]: Moving exported symbols will lead to incompatibility to existing projects and should be conducted only when necessary.

File: README.md [hunk 1]
```diff
@@ -268,6 +268,25 @@ The echo method uses the following configuration (0.3, 100) and other methods us
   }
 }
 ```
+##### Degradation: Category=degradation
+
+[JSON Schema](https://github.com/cloudwego/kitex/blob/develop/pkg/circuitbreak/item_circuit_breaker.go#L30)
+
+| Variable   | Introduction                       |
+|------------|------------------------------------|
```
Comment (felix021) [inline README.md:276]: `enabled` should also be documented.

File: pkg/degradation/item_degradation.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	c.config.Store(defaultConfig)
+	return c
+}
+
+// NotifyPolicyChange to receive policy when it changes
+func (c *Container) NotifyPolicyChange(cfg *Config) {
+	c.config.Store(cfg)
+}
+
+func (c *Container) GetAclRule() acl.RejectFunc {
```
Comment (felix021) [inline pkg/degradation/item_degradation.go:70]: Please add more unit tests to ensure the coverage of new/modified codes.

File: pkg/degradation/degradation.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+import (
+	"context"
+	"errors"
+
+	"github.com/cloudwego/kitex/pkg/endpoint"
+	"github.com/cloudwego/kitex/pkg/kerrors"
+)
+
+func NewDegradationMiddleware(container *Container) endpoint.Middleware {
```
Comment (felix021) [inline pkg/degradation/degradation.go]: cloudwego/kitex/pkg/acl 包下面的 `NewACLMiddleware` 已经有实现了，并且是 Kitex 的默认中间件，没有必要自己再实现一份

Comment (felix021) [reply pkg/degradation/degradation.go]: 只需要使用 `client.WithACLRules` 指定就行了。

File: pkg/degradation/item_degradation.go (hunk 2 above)
Comment (felix021) [reply pkg/degradation/item_degradation.go:70]: 这个文件的测试也请补充上

File: pkg/degradation/degradation.go (hunk 3 above)
Comment (CryoPhotonix) [reply pkg/degradation/degradation.go]: done

File: pkg/degradation/item_degradation.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	return func(ctx context.Context, request interface{}) (reason error) {
+		cfg := c.config.Load().(*Config)
+		if !cfg.Enable {
+			return nil
+		}
+		if fastrand.Intn(100) < cfg.Percentage {
+			return kerrors.ErrACL
```
Comment (felix021) [inline pkg/degradation/item_degradation.go]: 这里改成一个明确的错误信息，让用户知道是因为这个降级策略丢弃的流量

Comment (CryoPhotonix) [reply pkg/degradation/item_degradation.go]: done

**Review by felix021 — APPROVED** [review]

**Review by rogerogers — COMMENTED** [review]
贴错地方了。。

**Pull request merged**